lib/internal/modules/esm/worker.js

```diff
@@ -74,21 +65,12 @@ function releaseLock() {
       }
     }
 
-    try {
-      serializedResponse = serialize(response);
-      if (serializedResponse.byteLength > data.length) {
-        throw new ERR_OUT_OF_RANGE('serializedResponse.byteLength', `<= ${data.length}`, serializedResponse.byteLength);
-      }
-    } catch (exception) {
-      serializedResponse = serialize(exception);
-    }
-
     // Send the method response (or exception) to the main thread
     try {
-      TypedArrayPrototypeSet(data, serializedResponse);
+      syncCommPort.postMessage(response);
```

Comment:
I'm pretty sure the cause of the huge performance difference between `SharedArrayBuffer` and `postMessage()` is because with the SAB, it's a single shared memory space (so the transfer across threads is merely a context switch and nothing is moved/copied); but with `postMessage()`, when only the first arg is used, the response is copied (which is relatively expensive, on top of consuming double the space).

I think if you instead transfer the response via the [2nd param](https://developer.mozilla.org/en-US/docs/Web/API/MessagePort/postMessage#parameters), we'll see more similar (better) performance with `postMessage()`

Reply:
The 2nd param of `postMessage` can only be used with transferable objects. `response` is not transferable. I still haven't seen any benchmark or perf numbers so I'm not really inclined to talk about it just based on assumptions.